### PR TITLE
add reference list

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request implements the reference list as suggested in  #3258.

~~**Please do not merge** since this pull request is still not complete. 
But: feel free to already comment on the pull request: especially on sorting.~~

### CODE REVIEW

**Comment by PrometheusPi** [comment]
As a cross reference:  #3054
Since we now compile a quite long list we run into the [ugly issue](https://github.com/ComputationalRadiationPhysics/picongpu/issues/3054#issuecomment-530445274) @axel mentioned in that issue. 
@ComputationalRadiationPhysics/picongpu-developers what do you think:
:+1: one long list of all references with additional copies underneath each `*.rst` file
:-1: remove all references at the and of each `*.rst` file

**Comment by sbastrakov** [comment]
@PrometheusPi nice! Fyi the PR can't be merged while it is marked as draft on github.

**Comment by Anton-Le** [comment]
I think a good way would be to try and do it similar to a cohesive publication, with references in a large list at the end.
In each section of the documentation one would then _replace_ all references with links to the long list.

Pro: 

1. (Hopefully) no duplication
2. It would be a format accustomed from papers and theses.

Con:

1. Slightly increased maintenance cost due to the need of keeping the large list open when editing the documentation.
2. Potential for dead links if somebody deletes entries from the list.

**Comment by PrometheusPi** [comment]
@sbastrakov Thanks for the info - I chose the draft mode on purpose to avoid an accidental merge. 
So far only two (contradicting) votes are given: one pro and one con. 
@ComputationalRadiationPhysics/picongpu-developers please vote!

**Comment by ax3l** [comment]
>  Since we now compile a quite long list we run into the ugly issue @axel mentioned in that issue.

Problem solved: just use double `_` in `.rst` links if you want to get rid of the warning. that will create a non-named link.

File: docs/source/usage/reference.rst [hunk 1]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+In the following a list of references is given, sorted by topics, that describe PIConGPU as a software.
+References to publications of implemented methods that were deveolped by other groups are also given for completness.
+These references are marked by an asterix. 
+
+
+General code references
+"""""""""""""""""""""""
+
+ - First publication on PIConGPU [Burau2010]_
+ - Currently main reference [Bussmann2013]_
```
Comment (ax3l) [inline docs/source/usage/reference.rst:100]:
```suggestion
 - Currently main reference [Bussmann2013]_
 - Most up-to-date reference [Huebl2019]_
```

File: docs/source/usage/reference.rst [hunk 2]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+       *Electromagnetic Radiation from Relativistic Electrons as Characteristic Signature of their Dynamics*,
+       Diploma Thesis at Technische Universität Dresden & Helmholtz-Zentrum Dresden - Rossendorf for the German Degree "Diplom-Physiker" (2012),
+       https://doi.org/10.5281/zenodo.843510
+
+.. [Bussmann2013]
+       Bussmann, M. et al.
+       *Radiative signatures of the relativistic Kelvin-Helmholtz instability*
+       SC ’13 Proceedings of the International Conference for High Performance Computing, Networking, Storage and Analysis (pp. 5-1-5–12)
+       https://doi.org/10.1145/2503210.2504564
+
```
Comment (ax3l) [inline docs/source/usage/reference.rst:200]:
```suggestion

.. [Huebl2014TPS]
        Huebl, A. et al.
        *Visualizing the Radiation of the Kelvin-Helmholtz Instability*,
        IEEE Transactions on Plasma Science 42.10 (2014),
        https://doi.org/10.1109/TPS.2014.2327392

```

File: docs/source/usage/reference.rst [hunk 3]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+.. [Obst-Huebl2019]
+       Obst-Hübl, L.,
+       *Achieving optimal laser-proton acceleration through multi-parameter interaction control*,
+       PhD Thesis at Technische Universität Dresden & Helmholtz-Zentrum Dresden - Rossendorf (2019),
+       https://doi.org/10.5281/zenodo.3252952
+
+.. [Huebl2019]
+       Huebl, A.,
+       *Predictive Simulations of Laser-Particle Accelerators with Manycore Hardware*,
```
Comment (ax3l) [inline docs/source/usage/reference.rst:275]:
```suggestion
       *PIConGPU: Predictive Simulations of Laser-Particle Accelerators with Manycore Hardware*,
```

File: docs/source/usage/reference.rst [hunk 4]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+
+ - classical radiation: [Pausch2012]_, [Pausch2019]_
+ - phase space analysis: [Huebl2014]_
+ - beam emittance: [Rudat2019]_
+ - coherent transistion radiation (CTR): [Castens2019]_
+
+Visualization
+"""""""""""""
+
+ - first implementation:
```
Comment (ax3l) [inline docs/source/usage/reference.rst:143]:
```suggestion
 - Kelvin-Helmholtz: [Huebl2014TPS]_
 - first implementation: 
```

File: docs/source/usage/reference.rst [hunk 5]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+ - Laser-driven proton beam structering
+ - Hybrid laser-driven/beam-driven plasms acceleration 
+ - Radiation from LWFA (concept paper)
+   
+
+Astrophysics
+""""""""""""
+
+ - Simulating the Kelvin Helmholtz instabilty (KHI) [Bussmann2013]_
+ - Visualizing the KHI
```
Comment (ax3l) [inline docs/source/usage/reference.rst:82]:
```suggestion
 - Visualizing the KHI [Huebl2014TPS]_
```

File: docs/source/usage/reference.rst [hunk 6]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+	*Entwicklung und Überprüfung eines Photonenmodells für die Abstrahlung durch hochenergetische Elektronen*,
+	Diploma Thesis at Technische Universität Dresden & Helmholtz-Zentrum Dresden - Rossendorf for the German Degree "Diplom-Physiker" (2016),
+	https://doi.org/10.5281/zenodo.192116
+
+.. [Matthes2016]
+        Matthes, A.,
+	*In-Situ Visualisierung und Streaming von Plasmasimulationsdaten*,
+	Diploma Thesis at Technische Universität Dresden & Helmholtz-Zentrum Dresden - Rossendorf for the German Degree "Diplom-Informatiker" (2016),
+	https://doi.org/10.5281/zenodo.55329
+
```
Comment (ax3l) [inline docs/source/usage/reference.rst:224]:
```suggestion

.. [Huebl2017]
         Huebl, A. et al.
	*On the Scalability of Data Reduction Techniques in Current and Upcoming HPC Systems from an Application Perspective*,
	ISC High Performance Workshops 2017, LNCS 10524, pp. 15-29 (2017),
	https://doi.org/10.1007/978-3-319-67630-2_2

```

File: docs/source/usage/reference.rst [hunk 7]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+ - coherent transistion radiation (CTR): [Castens2019]_
+
+Visualization
+"""""""""""""
+
+ - first implementation: 
+
+Input/Output
+""""""""""""
+
```
Comment (ax3l) [inline docs/source/usage/reference.rst:147]:
```suggestion

- parallel HDF5, ADIOS1, compression, data reduction and I/O performance model [Huebl2017]_

```

File: docs/source/usage/reference.rst (hunk 4 above)
Comment (ax3l) [reply docs/source/usage/reference.rst:143]: I think our ISAAC 2016 ISC paper even won a best-paper award: https://arxiv.org/abs/1611.09048

File: docs/source/usage/reference.rst [hunk 8]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+Laser plasma acceleration
+"""""""""""""""""""""""""
+
+ - Laser wakefield acceleration (LWFA) with self-truncated ionization-injection (STII)
+ - Ion acceleration from droplet iradiation
+ - Laser-driven proton beam structering
+ - Hybrid laser-driven/beam-driven plasms acceleration 
+ - Radiation from LWFA (concept paper)
+
```
Comment (ax3l) [inline docs/source/usage/reference.rst:76]: Just dumping this quickly:
- Debus PRX 2019
- Huebl PPCF 2020
- Hilz Nat. Comm. 2018 (below: `Hilz2018`)
- Obst-Huebl Nat. Comm 2018
- Couperus Nat. Comm. 2017
- Obst Sci. Rep. 2017
- Schramm J. Phy.s Conf.Ser. 2017
- Fortmann-Grtoe: NOBUGS Conf. 2016, SPIE 2017

File: docs/source/usage/reference.rst (hunk 6 above)
Comment (PrometheusPi) [reply docs/source/usage/reference.rst:224]: @ax3l Thanks for commenting this pull request. But this is a work in progress draft and I did not continue to work on this due to the issue of double references _ I will tray your solution above and then will add further references - I would never forget your plasma-p*** publication.

File: docs/source/usage/reference.rst [hunk 9]
```diff
@@ -50,3 +50,240 @@ Thank you and enjoy PIConGPU and our community!
...
+QED code extensions
+"""""""""""""""""""
+
+Diagnostic methods
+""""""""""""""""""
+
+ - classical radiation: [Pausch2012]_, [Pausch2019]_
+ - phase space analysis: [Huebl2014]_
+ - beam emittance: [Rudat2019]_
+ - coherent transistion radiation (CTR): [Castens2019]_
```
Comment (n01r) [inline docs/source/usage/reference.rst:138]: missing **r** --> `[Carstens2019]_`

**Comment by psychocoderHPC** [comment]
@PrometheusPi Can we finish this PR? If some references are missing it is better than not have them in the codebase. We are short before the next release therefore we should finish it soon or it will be moved far into the future.

**Comment by PrometheusPi** [comment]
@psychocoderHPC good point, I will finish this up but will reference papers multiple times since there seems to be no consistent way to have a unified reference file.

**Comment by PrometheusPi** [comment]
The last commit is just another add-on. The list is yet not complete.

**Comment by PrometheusPi** [comment]
@psychocoderHPC and @sbastrakov do we have more recent publications showing benchmarks of PIConGPU?

File: docs/source/usage/reference.rst (hunk 8 above)
Comment (PrometheusPi) [reply docs/source/usage/reference.rst]: @ax3l I assume with *Schramm J. Phy.s Conf.Ser. 2017* you mean https://doi.org/10.1088/1742-6596/874/1/012028, is this correct? There are no PIConGPU results presented/discussed in there. Why did you want to reference this paper?

Comment (PrometheusPi) [reply docs/source/usage/reference.rst]: @ax3l reagrding the 
> Simulations of single-particle imaging of hydrated proteins with x-ray free-electron lasers
> Fortmann-Grote, C ; Bielecki, J ; Jurek, Z ; Santra, R ; Ziaja-Motyka, B ; Mancuso, A. P ; Sawhney, Kawal ; Chubar, Oleg 
> Proceedings of SPIE - The International Society for Optical Engineering, 2017-08-23, Vol.10388, p.103880M-103880M-22

I do not have access to this SPIE journal. Could you please describe how PIConGPU was used for this paper?

**Comment by PrometheusPi** [comment]
I found a solution to the reference issue: https://sphinxcontrib-bibtex.readthedocs.io/en/latest/index.html

**Comment by PrometheusPi** [comment]
@ComputationalRadiationPhysics/picongpu-maintainers Please feel free to review. Since there seems to be solution to have a central bibliography but keep local references on each page as a kind of short literature review of the topic (as it is currently), I just added all relevant references in `docs/source/usage/reference.rst`

**Comment by PrometheusPi** [comment]
Ping @ComputationalRadiationPhysics/picongpu-maintainers - ready for review.

File: docs/source/usage/reference.rst [hunk 10]
```diff
@@ -50,3 +50,475 @@ Thank you and enjoy PIConGPU and our community!
...
+Particle pushers
+""""""""""""""""
+ - Boris pusher\ :sup:`\*` [Boris1970]_
+ - Vay pusher\ :sup:`\*` [Vay2008]_
+ - Reduces Landau-Lifshitz pusher\ :sup:`\*` [Vranic2016]_
+
+Current deposition
+""""""""""""""""""
+ - Esirkepov method\ :sup:`\*` [Esirkepov2001]_
+ - Villasenor and Buneman method\ :sup:`\*` [Villasenior1991]_
```
Comment (sbastrakov) [inline docs/source/usage/reference.rst:132]: @PrometheusPi is the idea of this section to reference all e.g. pushers and field solvers we implement, or only "most important" ones?

Comment (PrometheusPi) [reply docs/source/usage/reference.rst:132]: @sbastrakov It would be ideal to reference all pushers/algorithms that we use.

Comment (sbastrakov) [reply docs/source/usage/reference.rst:132]: Forgot `HigueraCary` particle pusher

Comment (PrometheusPi) [reply docs/source/usage/reference.rst]: @sbastrakov I added the Higuera-Cary pusher.

File: docs/source/usage/reference.rst [hunk 11]
```diff
@@ -50,3 +50,475 @@ Thank you and enjoy PIConGPU and our community!
...
+	https://doi.org/10.1038/s41467-021-23000-7
+
+.. [Koehler2021]
+        Koehler, A., Pausch, R., Bussmann, M., et al.,
+	*Restoring betatron phase coherence in a beam-loaded laser-wakefield accelerator*,
+	Phys. Rev. Accel. Beams 24, 091302 – 20 September 2021,
+	https://doi.org/10.1103/PhysRevAccelBeams.24.091302
+
+List of other references in chronological order
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (PrometheusPi) [reply docs/source/usage/reference.rst:405]: The list above shows all papers where PIConGPU was used (either for simulation or to implement a method).
The second "other" list are publications not related to PIConGPU originally, but where we "extracted" methods from.

[the remark this answers is not in the document]
Comment (PrometheusPi) [reply docs/source/usage/reference.rst:405]: I would favour leafing it this way.

**Comment by PrometheusPi** [comment]
@sbastrakov Are you fine with the changes? Could you review and merge this pull request?

**Review by sbastrakov — APPROVED** [review]

**Comment by PrometheusPi** [comment]
Thanks @sbastrakov for your review :+1:

File: docs/source/usage/reference.rst (hunk 8 above)
Comment (ax3l) [reply docs/source/usage/reference.rst]: Oh, must have listed by accident. I wasn't sure anymore what we added in the J. Phy.s Conf.Ser., thanks for checking.

Proceedings of SPIE: also listed by accident, that's a simex_platform follow-up paper unrelated to PIConGPU.

**Pull request merged**